src/client/Client.cc

```diff
@@ -4595,12 +4596,22 @@ int Client::check_permissions(Inode *in, int flags, int uid, int gid)
     }    
   }
 #endif
+  unsigned want = 0;
+  if ((flags & O_ACCMODE) == O_WRONLY)
+    want = 2;
+  else if ((flags & O_ACCMODE) == O_RDWR)
+    want = 6;
+  else if ((flags & O_ACCMODE) == O_RDONLY)
+    want = 4;
+
+  int ret = _posix_acl_permission(in, uid, gid, sgids, sgid_count, want);
+  if (ret != -EACCES)
```

Comment:
Hmm, so the logic here is that if posix_acl_permission says 0, we return 0 (allow access), but if the ACLs deny permission we fall back to check_mode.  Intuitively that seems backwards to me, shouldn't we require that *both* tests pass rather than *either*?  I need to read more about what the rules are meant to be